sdk/src/types/block/output/nft.rs

```diff
@@ -94,22 +94,27 @@ impl NftOutputBuilder {
         self
     }
 
-    ///
+    /// Adds an [`UnlockCondition`] to the builder.
     #[inline(always)]
-    pub fn add_unlock_condition(mut self, unlock_condition: UnlockCondition) -> Self {
-        self.unlock_conditions.push(unlock_condition);
+    pub fn add_unlock_condition(mut self, unlock_condition: impl Into<UnlockCondition>) -> Self {
+        self.unlock_conditions.push(unlock_condition.into());
         self
     }
 
-    ///
+    /// Sets a collection of [`UnlockCondition`] to the builder.
```

Comment:
It would probably be good to specify in the comment that this is a replacement, rather than an append. Same for the other variants of course.

Reply:
I was under the impression `Sets` would mean that, but I can clarify for sure